### PR TITLE
home-assistant: pythonPackages.zigpy-znp: init at 0.2.2; pythonPackages.zigpy: 0.22.2 -> 0.26.0; pythonPackages.zigpy-cc: 0.5.1 -> 0.5.2; pythonPackages.zigpy-xbee: 0.12.1 -> 0.13.0; pythonPackages.zigpy-zigate: 0.6.1 -> 0.6.2

### PR DESCRIPTION
**Pull request by mvnetbiz**
###### Motivation for this change
Home Assistant update broke ZHA component by adding a new zigpy module zigpy-znp.

###### Things done
Init package pythonPackages.zigpy-znp.
This requires updating zigpy, and the other zigpy python modules.

I have checked to make sure this works on my H-A server.

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101557` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>16 package built:</summary>

  - python37Packages.bellows
  - python37Packages.zha-quirks
  - python37Packages.zigpy
  - python37Packages.zigpy-cc
  - python37Packages.zigpy-deconz
  - python37Packages.zigpy-xbee
  - python37Packages.zigpy-zigate
  - python37Packages.zigpy-znp
  - python38Packages.bellows
  - python38Packages.zha-quirks
  - python38Packages.zigpy
  - python38Packages.zigpy-cc
  - python38Packages.zigpy-deconz
  - python38Packages.zigpy-xbee
  - python38Packages.zigpy-zigate
  - python38Packages.zigpy-znp
</details>

File: pkgs/development/python-modules/zigpy-cc/default.nix [hunk 1]
```diff
@@ -1,17 +1,17 @@
-{ stdenv, buildPythonPackage, fetchPypi
+{ python, stdenv, buildPythonPackage, fetchPypi
```
Comment (mweinelt) [inline pkgs/development/python-modules/zigpy-cc/default.nix:1]: What's the purpose of this change?

File: pkgs/development/python-modules/zigpy-znp/default.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
+{ stdenv
```
Comment (mweinelt) [inline pkgs/development/python-modules/zigpy-znp/default.nix:1]: Just importing `lib` would do.

**Review by mweinelt — CHANGES_REQUESTED** [review]
I noticed a few components were loosing support during the last upgrade and didn't investigate. I'm sorry that broke your setup, I'll be more careful from now on.

You seem to introduce one spurious change in `zigpy-cc`. Apart from that things LGTM.

Result of `nixpkgs-review pr 101557` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>16 packages built:</summary>
  <ul>
    <li>python37Packages.bellows</li>
    <li>python37Packages.zha-quirks</li>
    <li>python37Packages.zigpy</li>
    <li>python37Packages.zigpy-cc</li>
    <li>python37Packages.zigpy-deconz</li>
    <li>python37Packages.zigpy-xbee</li>
    <li>python37Packages.zigpy-zigate</li>
    <li>python37Packages.zigpy-znp</li>
    <li>python38Packages.bellows</li>
    <li>python38Packages.zha-quirks</li>
    <li>python38Packages.zigpy</li>
    <li>python38Packages.zigpy-cc</li>
    <li>python38Packages.zigpy-deconz</li>
    <li>python38Packages.zigpy-xbee</li>
    <li>python38Packages.zigpy-zigate</li>
    <li>python38Packages.zigpy-znp</li>
  </ul>
</details>

File: pkgs/development/python-modules/zigpy-cc/default.nix (hunk 1 above)
Comment (mvnetbiz) [reply pkgs/development/python-modules/zigpy-cc/default.nix]: I was trying to add testing, but Pypi tarballs don't have the tests, and I just forgot to remove that afterwards, it's fixed now.

File: pkgs/development/python-modules/zigpy-znp/default.nix (hunk 2 above)
Comment (mvnetbiz) [reply pkgs/development/python-modules/zigpy-znp/default.nix:1]: I reverted the other change, is this fine to leave?

File: pkgs/development/python-modules/zigpy-cc/default.nix (hunk 1 above)
Comment (mweinelt) [reply pkgs/development/python-modules/zigpy-cc/default.nix]: I was really expecting tests to run, since this has checkInputs set. Can you move to `fetchFromGitHub` instead? We need either working tests (preferred) or `pythonImportTests`.

File: pkgs/development/python-modules/zigpy-znp/default.nix (hunk 2 above)
Comment (mweinelt) [reply pkgs/development/python-modules/zigpy-znp/default.nix:1]: It's just an unnecessary indirection. I'm not aware of any harm it does.

**Review by mweinelt — CHANGES_REQUESTED** [review]
I kinda expected tests to be present in all those packages and didn't verify that before, since `checkInputs` were given and the build passed. But all of these packages show 

```
----------------------------------------------------------------------
Ran 0 tests in 0.000s
```

and they also don't have a `pythonImportsCheck` set up, which is so helpful, since python code can break so easily at runtime. So my recommendation is:

1. Try to get tests working
2. If they are not packaged in PyPi fetch from Git instead
3. If there are no tests or they don't work at all, set `doCheck = false;`
4. Set up `pythonImportsCheck` instead

**Review by etu — APPROVED** [review]
I've tested these changes on my Home assistant server with my zigbee card. With these changes my Zigbee works again! I've kinda... ignored it failing for a bit.

But it's still worth to look into the tests mentioned by @mweinelt

File: pkgs/development/python-modules/zigpy-cc/default.nix [hunk 3]
```diff
@@ -1,22 +1,41 @@
...
-  meta = with stdenv.lib; {
+  propagatedBuildInputs = [
+    pyserial
+    pyserial-asyncio
+    zigpy
+  ];
+
+  checkInputs = [
+    asynctest
+    pytest
```
Comment (mweinelt) [inline pkgs/development/python-modules/zigpy-cc/default.nix:30]: Use pytestCheckHook instead and drop the `checkPhase = "pytest";`.

File: pkgs/development/python-modules/zigpy-xbee/default.nix [hunk 4]
```diff
@@ -1,20 +1,39 @@
...
-  meta = with stdenv.lib; {
+  buildInputs = [
+    pyserial
+    pyserial-asyncio
+    zigpy
+  ];
+
+  checkInputs = [
+    asynctest
+    pytest
```
Comment (mweinelt) [inline pkgs/development/python-modules/zigpy-xbee/default.nix:30]: as above

File: pkgs/development/python-modules/zigpy-zigate/default.nix [hunk 5]
```diff
@@ -1,22 +1,41 @@
...
-  meta = with stdenv.lib; {
+  buildInputs = [
+    pyserial
+    pyserial-asyncio
+    zigpy
+  ];
+
+  checkInputs = [
+    asynctest
+    pytest
```
Comment (mweinelt) [inline pkgs/development/python-modules/zigpy-zigate/default.nix:30]: as above

File: pkgs/development/python-modules/zigpy-znp/default.nix [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    pyserial
+    pyserial-asyncio
+    voluptuous
+    zigpy
+  ];
+
+  checkInputs = [
+    asynctest
+    coveralls
+    pytest
```
Comment (mweinelt) [inline pkgs/development/python-modules/zigpy-znp/default.nix:41]: as above

File: pkgs/development/python-modules/zigpy/default.nix [hunk 7]
```diff
@@ -1,20 +1,46 @@
...
+    aiohttp
+    crccheck
+    pycrypto
+    pycryptodome
+    voluptuous
+  ];
+
+  checkInputs = [
+    asynctest
+    pytest
```
Comment (mweinelt) [inline pkgs/development/python-modules/zigpy/default.nix:36]: as above

**Review by mweinelt — CHANGES_REQUESTED** [review]
Thanks, the tests are all looking to be working! I have two remaining requests:

Using `pytestCheckHook` is advantageous if you are using `pytest`, since it allows you to declaratively manage the `checkPhase` with `disabledTests` and `pytestFlagsArray` in the future. You also don't have to specify the `checkPhase` that way.

Please replace `pytest` with `pytestCheckHook` everywhere and drop all `checkPhase`s that just call `pytest`.

Regarding the commit messages: These packages are python3 only, so their attribute prefix should rather be `python3Packages`.

```
❯ nix-build -A pythonPackages.zigpy
error: asynctest-0.13.0 not supported for interpreter python2.7
```

**Comment by mvnetbiz** [comment]
It appears the tests are broken for python 3.7.8, possibly due to https://github.com/Martiusweb/asynctest/issues/152. Should I set doCheck false for python 3.7.8?

**Comment by mweinelt** [comment]
On Mon, Oct 26, 2020 at 04:52:12PM -0700, Matt Votava wrote:
> It appears the tests are broken for python 3.7.8, possibly due to https://github.com/Martiusweb/asynctest/issues/152. Should I set doCheck false for python 3.7.8?
> 

As these packages are exclusively used in home-assistant and that runs
on python 3.8, go ahead and disable them for 3.7.

**Comment by mvnetbiz** [comment]
I just want to check, disable the whole package, or just the tests? I disabled the package now. If that is what you meant should I change the commit message to python38Packages?

**Comment by mweinelt** [comment]
As the package does not seem to work >3.7.8 I don't really care if we disable the package or the tests.

python3Packages are currently synonym for python38Packages, so the former is fine.

**Review by mweinelt — APPROVED** [review]

**Comment by mweinelt** [comment]
Thanks!
